server/atlassian_connect.go

```diff
@@ -50,6 +51,26 @@ func httpACInstalled(p *Plugin, w http.ResponseWriter, r *http.Request) (int, er
 			errors.WithMessage(err, "failed to unmarshal request")
 	}
 
+	// Only allow this operation once, a JIRA instance must already exist
+	// for asc.BaseURL but its EventType would be empty.
+	ji, err := p.LoadJIRAInstance(asc.BaseURL)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to load instance "+asc.BaseURL)
+	}
+	if ji == nil {
+		return http.StatusNotFound,
+			errors.New(fmt.Sprintf("Jira instance %s must first be added to Mattermost", asc.BaseURL))
```

Comment:
alternative:
`fmt.Errorf("Jira instance %s must first be added to Mattermost", asc.BaseURL)`

Reply:
I converted (almost) everything to `errors.New` and `errors.Errorf` for consistency. We should probably have a separate conversation on tradeoffs and the best practices of using simple errors (`fmt.Errorf`, `errors.WithMessage`) vs those embedding the call stacks (`errors.New/Errorf`, `errors.Wrap`). I am 0/5 just would like consistency.